controller/contour/controller.go

```diff
@@ -247,3 +251,16 @@ func envoyDaemonSetPodSelector(contour *operatorv1alpha1.Contour) *metav1.LabelS
 		},
 	}
 }
+
+// contourFinalized returns true if contour is finalized.
+func contourFinalized(contour *operatorv1alpha1.Contour) bool {
+	if contour.Finalizers != nil {
```

Comment:
nit: you shouldn't need to nil-check here, since range over a nil slice won't panic